tests/datasets/test_events_processing.py

```diff
@@ -10,15 +10,7 @@
 
 
 def test_events_processing() -> None:
-    query_body = {
-        "selected_columns": [
-            [
-                "arrayElement",
-                ["tags.value", "indexOf", ["tags.key", "'transaction'"]],
-                "tags[transaction]",
-            ]
-        ]
-    }
```

Comment:
Did the purpose of this test change? The before and after of this change seem quite a bit different. I'd think the original behavior would still be expected, but not sure if this query body was structured this way intentionally or incidentally.

Reply:
It did change, because, I wanted to test promotion logic in the query processing, though, since I did not have the translation in place yet, tags resolution from `tags[key]` was not happening so I had to fake it. 

I am experimenting if a test like this to test a dataset processing end to end (to replace eventually test_api) works. 
To me there should be a very small number of very stable integration tests for each dataset that test the basic processing cases (ideally 5-10 cases). I think tags resolution/promotion is one of those for events.